### PR TITLE
Do not enumerate types from non-extension DLLs.

### PR DESCRIPTION
**Pull request by zachnew-msft**
Not all DLLs reference the SDK. If a DLL does not reference the SDK, then
there is no way that said DLL can contain a plugin or extension. This change
will cause the extension discovery to skip over any DLLs that do not reference
the SDK when doing Type enumeration.

After validating that the DLL is acceptable according to the validator, it is
loaded and passed to ProcessAssembly. Here, we add a check for an SDK reference
and return early if the reference is not present.

Tests have been added for this scenario.

### CODE REVIEW

File: src/Microsoft.Performance.Testing/FakeAssembly.cs [hunk 1]
```diff
@@ -12,23 +11,35 @@ public sealed class FakeAssembly
...
+        public override AssemblyName GetName()
         {
-            Assert.IsNotNull(this.TypesToReturn);
+            return this.AssemblyName;
+        }
 
-            return this.TypesToReturn;
+        public override AssemblyName[] GetReferencedAssemblies()
+        {
+            return this.ReferencedAssemblies ?? Array.Empty<AssemblyName>();
```
Comment (jmaxson-ms) [inline src/Microsoft.Performance.Testing/FakeAssembly.cs:37]: >?? Array.Empty<AssemblyName>(); [](start = 45, length = 31)

why is this needed?

**Review by jmaxson-ms — APPROVED** [review]
:shipit:

File: src/Microsoft.Performance.Testing/FakeAssembly.cs (hunk 1 above)
Comment (jmaxson-ms) [reply src/Microsoft.Performance.Testing/FakeAssembly.cs:37]: asking for my own benefit. if it's not needed, i'm not suggestion you need to change it. just curious if there's a path I'm not thinking of.

**Review by trgibeau — APPROVED** [review]
